src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpec.scala

```diff
@@ -459,82 +449,81 @@ class UserServiceSpec extends AnyFlatSpec with Matchers with TestSupport with Mo
 
   it should "return conflict when there's an existing subject for a given azureB2CId" in{
     val user = genWorkbenchUserAzure.sample.get
-    dirDAO.createUser(WorkbenchUser(user.id,  user.googleSubjectId, user.email, user.azureB2CId), samRequestContext).unsafeRunSync()
+    dirDAO.createUser(user, samRequestContext).unsafeRunSync()
     val exception = intercept[WorkbenchExceptionWithErrorReport] {
       service.registerUser(user, samRequestContext).unsafeRunSync()
     }
     exception.errorReport shouldEqual ErrorReport(StatusCodes.Conflict, s"user ${user.email} already exists")
   }
 
   "UserService inviteUser" should "create a new user" in{
-    val user = genInviteUser.sample.get
-    service.inviteUser(user, samRequestContext).unsafeRunSync()
-    val res = dirDAO.loadUser(user.inviteeId, samRequestContext).unsafeRunSync()
-    val registrationRes = registrationDAO.loadUser(user.inviteeId, samRequestContext).unsafeRunSync()
-    res shouldBe Some(WorkbenchUser(user.inviteeId, None, user.inviteeEmail, None))
+    val userEmail = genNonPetEmail.sample.get
+    service.inviteUser(userEmail, samRequestContext).unsafeRunSync()
+    val userId = dirDAO.loadSubjectFromEmail(userEmail, samRequestContext).unsafeRunSync().value.asInstanceOf[WorkbenchUserId]
+    val res = dirDAO.loadUser(userId, samRequestContext).unsafeRunSync()
+    val registrationRes = registrationDAO.loadUser(userId, samRequestContext).unsafeRunSync()
+    res shouldBe Some(SamUser(userId, None, userEmail, None, false))
     registrationRes shouldEqual res
   }
 
   it should "reject blocked domain" in {
-    val user = genInviteUser.sample.get
     intercept[WorkbenchExceptionWithErrorReport] {
-      service.inviteUser(user.copy(inviteeEmail = WorkbenchEmail(s"user@$blockedDomain")), samRequestContext).unsafeRunSync()
+      service.inviteUser(WorkbenchEmail(s"user@$blockedDomain"), samRequestContext).unsafeRunSync()
     }.errorReport.statusCode shouldBe Some(StatusCodes.BadRequest)
   }
 
-  it should "return conflict when there's an existing subject for a given userId" in{
-    val user = genInviteUser.sample.get
-    val email = genNonPetEmail.sample.get
-    dirDAO.createUser(WorkbenchUser(user.inviteeId, None, email, None), samRequestContext).unsafeRunSync()
-    val res = service.inviteUser(user, samRequestContext).attempt.unsafeRunSync().swap.toOption.get.asInstanceOf[WorkbenchExceptionWithErrorReport]
-    Eq[WorkbenchExceptionWithErrorReport].eqv(res, new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"identity with id ${user.inviteeId} already exists"))) shouldBe true
-  }
```

Comment:
why don't we need a test like this anymore?

Reply:
I changed the `inviteUser` from taking an object that included both user id and email to just taking email so this case became impossible.